python/src/aac/io/parser/_yaml.py

```diff
@@ -68,3 +103,16 @@ def assert_definition_has_name(model):
     # Raise an error if any of the loaded YAML models are incomplete.
     models_without_imports = list(filter(lambda m: not is_import(m), models))
     all(map(assert_definition_has_name, models_without_imports))
+
+
+def _yaml_error_logging(error_type, error, source, content):
+    logging.error(f"Encountered YAML {error_type} Error-Problem: {error.problem}. Encountered in: {source} at {error.context}")
+    if error.problem_mark:
+        logging.error(f"Error occurs at line, column: {error.problem_mark.line+1}, {error.problem_mark.column+1}")
+    logging.error(f"Content of error: {content}")
+
+
+def _yaml_raise_error(error_type, error, source):
+    raise ParserError(source, [f"Encountered an invalid YAML with the following {error_type} error: {error.problem}",
+                               f"Encountered error at line, column: {error.problem_mark.line+1}, {error.problem_mark.column+1}",
+                               f"Context of the error: {error.context}"]) from None
```

Comment:
We should put the actual string(s)/message(s) as a return value from the function so that we can re-use the error messages in the logging and `ParserError`. The content of the messages are virtually identical.

Reply:
Think I got this split apart in a better way this time.

Reply:
Looks good. I'll accept this on Cameron's behalf.